liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/CoreToLogic.hs

```diff
@@ -2,7 +2,27 @@ module Language.Haskell.Liquid.GHC.CoreToLogic where
 
 coreToLogic :: String
 coreToLogic = unlines
-  [ "define Data.Set.Base.singleton x      = (Set_sng x)"
+  [ "define GHC.Types.True                   = (true)"
+  , "define GHC.Internal.Real.div x y        = (x / y)"
+  , "define GHC.Internal.Real.mod x y        = (x mod y)"
+  , "define GHC.Internal.Real.fromIntegral x = (x)"
+
+  , "define GHC.Classes.not x                = (~ x)"
+  , "define GHC.Internal.Base.$ f x          = (f x)"
+  , ""
+  , "define GHC.CString.unpackCString# x = x"
+  , ""
+  , "define Main.mempty       = (mempty)"
+  , ""
+  , "define Control.Parallel.Strategies.withStrategy s x = (x)"
+  , ""
+  , "define Language.Haskell.Liquid.Equational.eq x y = (y)"
+
+  , "define Language.Haskell.Liquid.ProofCombinators.cast x y = (y)"
+  , "define Liquid.ProofCombinators.cast x y = (y)"
+  , "define ProofCombinators.cast x y = (y)"
+  , ""
+  , "define Data.Set.Base.singleton x      = (Set_sng x)"
```

Comment:
The defines for set operations could be moved to `src/Data/Set_LHAssumptions.hs` perhaps.